packages/blob/src/helpers.ts

```diff
@@ -83,3 +107,108 @@ export function isPlainObject(value: unknown): boolean {
 }
 
 export const disallowedPathnameCharacters = ['#', '?', '//'];
+
+// Chrome: implemented https://developer.chrome.com/docs/capabilities/web-apis/fetch-streaming-requests
+// Microsoft Edge: implemented (Chromium)
+// Firefox: not implemented, BOO!! https://bugzilla.mozilla.org/show_bug.cgi?id=1469359
+// Safari: not implemented, BOO!! https://github.com/WebKit/standards-positions/issues/24
+export const supportsRequestStreams = (() => {
+  let duplexAccessed = false;
+
+  const hasContentType = new Request(getApiUrl(), {
+    body: new ReadableStream(),
+    method: 'POST',
+    // @ts-expect-error -- duplex option exists
+    get duplex() {
+      duplexAccessed = true;
+      return 'half';
+    },
+  }).headers.has('Content-Type');
+
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition -- duplexAccessed may be true here
+  return duplexAccessed && !hasContentType;
+})();
+
+export function getApiUrl(pathname = ''): string {
+  let baseUrl = null;
+  try {
+    // wrapping this code in a try/catch as this function is used in the browser and Vite doesn't define the process.env.
+    // As this varaible is NOT used in production, it will always default to production endpoint
+    baseUrl =
+      process.env.VERCEL_BLOB_API_URL ||
+      process.env.NEXT_PUBLIC_VERCEL_BLOB_API_URL;
+  } catch {
+    // noop
+  }
+  return `${baseUrl || 'https://blob.vercel-storage.com'}${pathname}`;
+}
+
+const TEXT_ENCODER =
+  typeof TextEncoder === 'function' ? new TextEncoder() : null;
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any -- We want to feature detect
+export function computeBodyLength(body: any): number {
+  /* eslint-disable @typescript-eslint/no-unsafe-member-access -- We want to feature detect */
+
+  if (typeof body === 'string') {
+    if (TEXT_ENCODER) {
+      return TEXT_ENCODER.encode(body).byteLength;
+    }
+
+    // React Native doesn't have TextEncoder
+    return new Blob([body]).size;
+  } else if (typeof body.byteLength === 'number') {
```

Comment:
no need for `else if` because all branches exit early

Reply:
Good, updated, I wish this was linted too